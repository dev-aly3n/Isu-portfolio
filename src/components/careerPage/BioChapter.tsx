import { useState } from "react";
import { motion } from "framer-motion";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleRight } from "@fortawesome/free-solid-svg-icons";

interface props {
  chap: { title: string; desc: string };
  firstChap: boolean;
}

const BioChapter: React.FC<props> = ({ chap, firstChap }) => {
  const [isExpanded, setIsExpanded] = useState(firstChap ? true : false);
  return (
    <motion.div
      layout
      className="flex flex-col justify-center items-center w-10/12 sm:w-full my-3"
    >
      <motion.h3
        layout
        onClick={() => setIsExpanded((prev) => !prev)}
        className="text-2xl font-bold self-start pl-5 cursor-pointer"
      >
        <FontAwesomeIcon className={isExpanded ? "rotate-90":"rotate-0"} icon={faAngleRight} />{" "}{chap.title}
      </motion.h3>
      <motion.div layout className="w-full h-px bg-white my-2" />
      <motion.p
        layout
        className={`text-base first-letter:text-primary-200 first-letter:font-semibold first-letter:text-5xl overflow-hidden
       ${isExpanded ? "" : "h-0"}`}
      >
        {chap.desc}
      </motion.p>
    </motion.div>
  );
};

export default BioChapter;
