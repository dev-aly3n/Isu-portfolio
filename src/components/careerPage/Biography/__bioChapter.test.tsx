import { render, screen } from "@testing-library/react";
import React, { useRef } from "react";
import { RefObject } from "react";
import BioChapter from "./BioChapter";


const chap = {
    title: "this is chapter 1",
    desc: "the chapter 1 is a test that we didn't implement yet but we will do in the short future."
}

describe("career/bioChapter", () => {
  it("snap", () => {
    const { container } = render(<BioChapter chap={chap} firstChap={true} />);
    expect(container).toMatchSnapshot();
  });
});
