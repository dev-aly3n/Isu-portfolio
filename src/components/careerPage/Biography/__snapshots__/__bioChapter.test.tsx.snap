// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/bioChapter snap 1`] = `
<div>
  <div
    class="bio-chapter-container"
    style="opacity: 0;"
  >
    <h3>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 rotate-90 duration-500"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
       
      this is chapter 1
    </h3>
    <div />
    <p
      class=" "
    >
      the chapter 1 is a test that we didn't implement yet but we will do in the short future.
    </p>
  </div>
</div>
`;
